tools/driver-cxx/driver-cxx.cpp

```diff
@@ -306,26 +308,36 @@ int main(int argc, char **argv) {
 
   auto sandbox = GetProcessSandbox(SandboxOption);
 
+  std::vector<std::shared_ptr<mull::Filter>> filterStorage;
   mull::Filters filters;
 
-  mull::NoDebugInfoFilter noDebugInfoFilter;
-  filters.mutationFilters.push_back(&noDebugInfoFilter);
-  filters.functionFilters.push_back(&noDebugInfoFilter);
-  filters.instructionFilters.push_back(&noDebugInfoFilter);
+  std::shared_ptr<mull::NoDebugInfoFilter> noDebugInfoFilter(
+      new mull::NoDebugInfoFilter);
+  std::shared_ptr<mull::FilePathFilter> filePathFilter(
+      new mull::FilePathFilter);
+
+  filterStorage.push_back(noDebugInfoFilter);
+  filterStorage.push_back(filePathFilter);
+
+  filters.mutationFilters.push_back(noDebugInfoFilter.get());
+  filters.functionFilters.push_back(noDebugInfoFilter.get());
+  filters.instructionFilters.push_back(noDebugInfoFilter.get());
 
-  mull::FilePathFilter filePathFilter;
-  filters.mutationFilters.push_back(&filePathFilter);
-  filters.functionFilters.push_back(&filePathFilter);
+  filters.mutationFilters.push_back(filePathFilter.get());
+  filters.functionFilters.push_back(filePathFilter.get());
 
   for (const auto &regex : ExcludePaths) {
-    filePathFilter.exclude(regex);
+    filePathFilter->exclude(regex);
   }
   for (const auto &regex : IncludePaths) {
-    filePathFilter.include(regex);
+    filePathFilter->include(regex);
   }
 
   if (junkDetectionEnabled) {
-    filters.mutationFilters.push_back(&junkFilter);
+    std::shared_ptr<mull::JunkMutationFilter> junkFilter(
```

Comment:
This is a kind of change I will need for white search as well: junk filter can now be created inside a scope of `if (junkDetectionEnabled)` and it will survive the scope and will live as long as the filter storage.